pkg/roachpb/api.proto

```diff
@@ -698,6 +698,43 @@ message DeprecatedVerifyChecksumResponse {
   optional ResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
 }
 
+enum ExportStorageProvider {
+  Unknown = 0;
+  LocalFile = 1;
+  //Http = 2;
```

Comment:
I went ahead and added a basic impl of HTTP -- I'm sure we'll need to flesh it out some more when we test against an actual webHDFS endpoint and other off-the-shelf webdav servers, but it seems to work against my dummy test server.